k9mail-library/src/main/java/com/fsck/k9/mail/store/imap/ImapFolder.java

```diff
@@ -919,8 +840,13 @@ public void fetchPartWithProgressCallback(Message message, Part part, MessageRet
                             String contentTransferEncoding =
                                     part.getHeader(MimeHeader.HEADER_CONTENT_TRANSFER_ENCODING)[0];
                             String contentType = part.getHeader(MimeHeader.HEADER_CONTENT_TYPE)[0];
-                            MimeMessageHelper.setBody(part, MimeUtility.createBodyWithProgressCallback(bodyStream, contentTransferEncoding,
-                                    contentType, progressCallback));
+                            if(progressCallback != null) {
```

Comment:
Yes, you're right, there is no need to check the null condition. Removing it now.